crates/prune/src/pruner.rs

```diff
@@ -87,14 +92,19 @@ impl<DB: Database> Pruner<DB> {
         // TODO(alexey): prune snapshotted segments of data (headers, transactions)
         let highest_snapshots = *self.highest_snapshots_tracker.borrow();
 
-        // Multiply `delete_limit` (number of row to delete per block) by number of blocks since
-        // last pruner run. `previous_tip_block_number` is close to `tip_block_number`, usually
-        // within `self.block_interval` blocks, so `delete_limit` will not be too high. Also see
-        // docs for `self.previous_tip_block_number`.
-        let mut delete_limit = self.delete_limit *
-            self.previous_tip_block_number
-                .map_or(1, |previous_tip_block_number| tip_block_number - previous_tip_block_number)
-                as usize;
+        // Multiply `self.delete_limit` (number of row to delete per block) by number of blocks
+        // since last pruner run. `self.previous_tip_block_number` is close to
+        // `tip_block_number`, usually within `self.block_interval` blocks, so
+        // `delete_limit` will not be too high. If it's too high, we additionally limit it by
+        // `self.prune_max_blocks_per_run`.
+        //
+        // Also see docs for `self.previous_tip_block_number`.
+        let blocks_since_last_run =
+            (self.previous_tip_block_number.map_or(1, |previous_tip_block_number| {
+                (tip_block_number - previous_tip_block_number) as usize
```

Comment:
should we use saturating here?

Reply:
it shouldn't be possible due to https://github.com/paradigmxyz/reth/blob/3218a3000c5cf8233f4bfae24dcf3e1c0f738faf/crates/prune/src/pruner.rs#L242-L263, which runs before `Pruner::run`, but fixed anyway just in case